tests/unit/module_utils/test_elbv2.py

```diff
@@ -159,3 +161,133 @@ def test_get_elb_attributes(self):
         actual_elb_attributes = self.elbv2obj.get_elb_attributes()
         # Assert we got the expected result
         assert actual_elb_attributes == expected_elb_attributes
+
+
+class TestELBListeners:
+    DEFAULT_PORT = 80
+    DEFAULT_PROTOCOL = "TCP"
+
+    def createListener(self, **kwargs):
+        result = {"Port": self.DEFAULT_PORT, "Protocol": self.DEFAULT_PROTOCOL}
+        if kwargs.get("port"):
+            result["Port"] = kwargs.get("port")
+        if kwargs.get("protocol"):
+            result["Protocol"] = kwargs.get("protocol")
+        if kwargs.get("certificate_arn") and kwargs.get("protocol") in ("TLS", "HTTPS"):
+            result["Certificates"] = [{"CertificateArn": kwargs.get("certificate_arn")}]
+        if kwargs.get("sslPolicy") and kwargs.get("protocol") in ("TLS", "HTTPS"):
+            result["SslPolicy"] = kwargs.get("sslPolicy")
+        if kwargs.get("alpnPolicy") and kwargs.get("protocol") == "TLS":
+            result["AlpnPolicy"] = kwargs.get("alpnPolicy")
+        return result
+
+    @pytest.mark.parametrize("current_protocol", ["TCP", "TLS", "UDP"])
+    @pytest.mark.parametrize(
+        "current_alpn,new_alpn",
+        [
+            (None, "None"),
+            (None, "HTTP1Only"),
+            ("HTTP1Only", "HTTP2Only"),
```

Comment:
Should we add a case here where `current_alpn` and `new_alpn` are the same, just to be thorough?

Reply:
done